src/twisted/words/test/test_tap.py

```diff
@@ -24,37 +28,37 @@ def setUp(self):
         self.file.write(self.PASSWD_TEXT)
         self.file.flush()
 
-    def tearDown(self):
+    def tearDown(self) -> None:
         """
         Close the dummy user database.
         """
         self.file.close()
 
-    def test_hostname(self):
+    def test_hostname(self) -> None:
         """
         Tests that the --hostname parameter gets passed to Options.
         """
         opt = tap.Options()
         opt.parseOptions(["--hostname", "myhost"])
         self.assertEqual(opt["hostname"], "myhost")
 
-    def test_passwd(self):
+    def test_passwd(self) -> None:
         """
         Tests the --passwd command for backwards-compatibility.
         """
         opt = tap.Options()
         opt.parseOptions(["--passwd", self.file.name])
         self._loginTest(opt)
 
-    def test_auth(self):
+    def test_auth(self) -> None:
         """
         Tests that the --auth command generates a checker.
         """
         opt = tap.Options()
         opt.parseOptions(["--auth", "file:" + self.file.name])
         self._loginTest(opt)
```

Comment:
should this be 
```suggestion
        return self._loginTest(opt)
```

Reply:
It  should return a deferred... or inlineCallbacks :)

If you have time, please update these tests. thanks